nautobot/ipam/querysets.py

```diff
@@ -1,9 +1,56 @@
 import uuid
 
+import netaddr
+from django.db.models import ExpressionWrapper, IntegerField, F, QuerySet, Q, UUIDField
+from django.db.models.expressions import RawSQL
+from django.db.models.functions import Coalesce, Length
+
+from nautobot.ipam.constants import IPV4_BYTE_LENGTH, IPV6_BYTE_LENGTH
 from nautobot.utilities.querysets import RestrictedQuerySet
 
 
-class PrefixQuerySet(RestrictedQuerySet):
+class NetworkQuerySet(QuerySet):
+    def net_equals(self, prefix):
+        return self.filter(
+            prefix_length=prefix.prefixlen,
+            network=bytes(prefix.network),
+            broadcast=bytes(prefix.broadcast),
+        )
+
+    def net_contained(self, prefix):
+        return self.filter(
+            prefix_length__gt=prefix.prefixlen,
+            network__gte=bytes(prefix.network),
+            broadcast__lte=bytes(prefix.broadcast),
```

Comment:
Explanatory comments would be helpful - for example, why are these not `network_gt` and `broadcast_lt` respectively?

Reply:
That's a "within" query innit?